Add Deye hybrid inverters

Blowfly69 opened the pull request:
Add templates for Deye Sun-(5-12)K-SG04LP3-EU and compatible Sunsynk hybrid inverters. Based on growatt-hybrid templates.

Fixes #7540

Blowfly69 committed:
Add Deye hybrid inverters

Add templates for Deye Sun-(5-12)K-SG04LP3-EU and compatible Sunsynk inverters

andig commented:
Seems a 

    make docs

is missing

premultiply committed:
add linebreak

premultiply committed:
remove scale

Blowfly69 commented on templates/definition/meter/deye-hybrid.yaml:
```diff
@@ -0,0 +1,124 @@
+products:
+  - brand: Deye
+    description:
+      generic: SUN-(5-12)K-SG04LP3-EU series hybrid inverter
```
Die Beschreibung ist da ggf. noch nicht ganz klar. Er gilt zumindest für alle Typen Deye SUN- X K-SG04LP3-EU, wobei X die Werte 5,6,8,10,12 annehmen kann. Darüber hinaus gilt er vermutlich auch für die Sunsynk Inverter (Tochterfirma von Deye) mit der ähnlichen Bezeichnung SYNK-8/10/12K-SG04LP3, die hardwaremäßig wohl fast identisch sind, aber eine andere UI haben.

Blowfly69 replied:
Die Sunsynk Konfiguration müsste aber halt jemand testen, der einen hat...

Blowfly69 replied:
Gibts eigentlich irgendwo Doku, wie die Templates funktionieren? Habe mich da etwas durchbeißen müssen und sicher noch nicht alles verstanden. Nach meinem Verständnis erzeugt der DOCS Pfad die Parameter, die während _evcc configure_ interaktiv abgefragt werden, und die dann im DEFINITION Pfad ins zugehörige Template eingesetzt werden? Wo findet man dann das finale YAML, oder existiert das nur intern?

Blowfly69 replied:
Soweit sind alle Änderungen drinnen. Bekomme jetzt beim Testen _evcc configure_ beim grid noch eine Fehlermeldung wg. fehlendem '-', pv und battery passen wohl. Was stimmt da nicht in der Syntax?
```
 In welchem Modus soll die Konfiguration durchgeführt werden? Fortgeschrittener Modus (Detailliertere Fragen, erfordert jedoch technisches Know-How)

 Was möchtest du machen? Eine neue evcc Konfigurationsdatei erstellen

- Hausinstallation einrichten

Wähle eines der folgenden PV Komplettsysteme aus, oder 'Mein Gerät ist nicht in der Liste' falls keines dieser Geräte vorhanden ist

 Wähle ein PV System: Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter

Führe folgende Einstellungen durch:

 Wähle die ModBus Schnittstelle aus TCP/IP
 Modbus ID [? for help] (1)

 Modbus ID 1
 IP-Adresse oder Hostname autoberry   
 Port 5021        

Teste die grid Konfiguration von Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter ...
   Fehler: yaml: line 23: did not find expected '-' indicator
 Der Test der grid Konfiguration von Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter ist fehlgeschlagen. Soll grid trotzdem in die Konfiguration aufgenommen werden? Yes
rden? (y/N) y
Teste die pv Konfiguration von Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter ...

Teste die battery Konfiguration von Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter ...

Deye SUN-(5/6/8/10/12)K-SG04LP3-EU series hybrid inverter wurde erfolgreich hinzugefügt.
```

Blowfly69 replied:
Fehler gefunden, war ein zu kleiner "indent" Wert definiert. Hat aber gebraucht, weil die Fehlermeldung auf Zeile 23 im yaml verwiesen hat, aber da wohl erst ab "render" gezählt wird...

Blowfly69 commented on templates/definition/meter/deye-hybrid.yaml:
```diff
@@ -0,0 +1,124 @@
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register: # manual non-sunspec register configuration
+      address: 520 # "Day_GridBuy_Power Wh"
```
Du meinst Summe aus gekauft und verkauft? Kann ich machen.

Blowfly69 replied:
522 Total_GridBuy_Power Wh low word  Scale 0.1kWh
523 Total_GridBuy_Power Wh high word Scale 0.1kWh
524 Total_GridSell_Power Wh low word Scale 0.1kWh
525 Total_GridSell_Power Wh_high word Scale 0.1kWh

Soll ich die beiden verrechnen (Buy-Sell?) oder welcher ist da der interessante?

premultiply replied:
Nur Import (Buy) ist relevant.

Blowfly69 commented on templates/definition/meter/deye-hybrid.yaml:
```diff
@@ -0,0 +1,124 @@
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register: # manual non-sunspec register configuration
+      address: 529 # "Day_PV_Power_Wh"
```
Bei PV verstehe ich "Total" nicht - nach meinem Verständnis enthält das Register die PV-Energie, die bisher am Tag generiert wurde?

Blowfly69 replied:
Das ist zumindest klar:
534  PV_power Total Wh_low word Scale 0.1kWh
535  PV_power Total Wh_high word Scale 0.1kWh

Blowfly69 committed:
Changed to Total energy registers

Total energy instead of day energy
description updated
scale 1.0 removed
line feed added

Blowfly69 committed:
Merge branch 'master' of https://github.com/Blowfly69/evcc

Blowfly69 commented on templates/definition/meter/deye-hybrid.yaml:
```diff
@@ -0,0 +1,123 @@
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register: # manual non-sunspec register configuration
+      address: 515 # "Today discharge of the battery"
```
Sollte das dann nicht auch noch auf /Total Energy/ umgestellt werden? Wenn ja, Total_Discharge oder Total_Charge?

premultiply replied:
Ja, unbedingt.
Total_Discharge

Blowfly69 committed:
Update deye-hybrid.yaml

Change Battery Energy to Total_Discharge

Blowfly69 committed:
inst

Blowfly69 committed:
Revert "inst"

This reverts commit ef7b3630b7cfde47ccba6da98aa98aa9d19fd327.

correct commit description

Blowfly69 committed:
Correct wrong indent value in template

Blowfly69 commented:
Auf Windows habe ich den Zugriff auf den Deye über ModBusProxy (hängt per RS485 an einem Raspi) jetzt erfolgreich getestet, evcc configure läuft ohne Fehler durch, evcc liest die Parameter des Deye m.E. korrekt aus. Die für evcc grundlegenden Funktionen des Deye bzgl. Überwachung von GRID-Anschluss und den drei möglichen PV-Strings sollten damit abgedeckt sein. Deye-Specials wie z.B. am GEN(erator)-Port angeschlossene (Mikro-/AC-)Wechselrichter müssen momentan noch manuell hinzufügt werden. Man könnte das vielleicht noch unter "advanced" in das Template packen, wüsste aber im Moment nicht wie das geht.

Blowfly69 committed:
Description (type name) corrected

Blowfly69 commented:
> Du müsstest die Änderungen an den beiden go-Files noch rückgängig machen. Das hat hier im PR nichts verloren.

Ok, ich weiß gar nicht wie die reingekommen sind... Muss ich den ganzen commit da rückgängig machen oder kann man die Files auch einzeln behandeln?

> Bzgl. GEN-Eingang: Wenns es geht einfach zu den PV-Strings addieren. Wenn nichts anliegt kommt da wahrscheinlich auch nur 0, oder?

Ja, prinzipiell schon, aber es gibt da auch noch andere Möglichkeiten, die man evtl. beim Konfigurieren abfragen müsste: Der GEN Port kann als Eingang für Generatoren und AC-gekoppelte WR, aber auch als "Smart Load"-Ausgang definiert werden, um z.B. bei PV-Überschuss einen Heizstab fürs Warmwasser anzuwerfen. Dann geht die Energie in die andere Richtung, und ich weiß nicht, ob und wie das in evcc korrekt berücksichtigt wird und ob der Deye das logisch korrekt behandelt (also Vorzeichen invertiert). Man müsste also abhängig davon dann evtl. die scale Faktoren negativ machen. Wenn das automatisch passt mit dem Vorzeichen, könnte man das so berücksichtigen wie du sagst, aber das müsste erst jemand mit Smart Load austesten - ich habe das nicht. Daher würde ich das erst mal rauslassen und ggf. ins Doc einen Hinweis schreiben, wie das umzusetzen ist.

Blowfly69 committed:
Undo changes to go.mod & go.sum

premultiply commented:
Jetzt sind noch mehr Änderungen an den go-Dateien drin.

Blowfly69 commented:
> Jetzt sind noch mehr Änderungen an den go-Dateien drin.

Ups, dann habe ich die falschen erwischt, kommt davon wenn man noch nie was mit Github gemacht hat... Ich dachte, ich hätte auch ein make docs vergessen, nachdem ich da was geändert hatte. Das muss jetzt leider bis Sonntag oder Montag warten, bin gerade unterwegs.

Blowfly69 committed:
next try to revert go.sum and go.mod

Blowfly69 committed:
Correct Sunsynk brand description (type)

Blowfly69 commented:
Nächster Versuch mit revert der beiden go files. Wenn das wieder nicht hinhaut, würde ich dich bitten @premultiply, das für mich zu machen...

premultiply commented:
Jetzt passt's! Danke!
